### ISSUE TITLE
Missing conversionEnvelope on iOS for ad conversions

### ISSUE THREAD

**Issue by Uni-verse** [brave/brave-ios#5059]
### Description: 
Missing ConversionEnvelope on iOS for ad conversions

### Steps to Reproduce 
  1. See Test Plan/Test Steps in https://github.com/brave/brave-core/pull/8640#issue-867524646

**Actual result:** 
conversionEnvelope not appended to confirmation payload. 

https://ads-serve.bravesoftware.com/v2/confirmation/261acb34-dde4-4255-9913-af8ec064e4bd/
{
"blindedPaymentTokens": ["zrO+trqwgPDShe2bG/NRC/mtrZdt8YdMatu2HqjjMBQ="],
"buildChannel": "dev",
"creativeInstanceId": "0e3fa44c-edd4-4ad3-b21e-e6c93ef001c4",
"payload": {},
"platform": "ios",
"publicKey": "JnPs9+7NSdmFflbIvMs0/6BBDt99JMgEvm+9mHNByjI=",
"studies": [],
"type": "conversion"
}

**Expected result:**
conversionEnvelope is appended to confirmation request

{
	"blindedPaymentTokens": ["tiQ/TRMLQhBXkrOAQMV6ydRGQ5P3EcEcd0vCcPC/jEE="],
	"buildChannel": "dev",
	"conversionEnvelope": {
		"alg": "crypto_box_curve25519xsalsa20poly1305",
		"ciphertext": "1FZEfhtpqAPx4u3sGHzrwI+7slPT/vt+//1wVE+ZeUdVrLGfE5LjqnM6BvSZkF+z",
		"epk": "CQqdjZAKOuMrWDImNqU9Nh8fx0VHNes8rVtsQlzT5i0=",
		"nonce": "5rl9XcaibA8hGhGyOBR566FeC8S/XEFI"
	},
	"creativeInstanceId": "0e3fa44c-edd4-4ad3-b21e-e6c93ef001c4",
	"payload": {},
	"platform": "ios",
	"publicKey": "JnPs9+7NSdmFflbIvMs0/6BBDt99JMgEvm+9mHNByjI=",
	"studies": [],
	"type": "conversion"
}

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easily

**Brave Version:** 
1.36.108

**Device details:** 
iPhone, iPad
iOS 15, iOS 14

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by btlechowski** [comment]
Steps:
1. Use catalog in charles: [5059.txt](https://github.com/brave/brave-ios/files/8200882/5059.txt)
1. Clean profile on dev 
1. Switch to staging
1. Restart browser
1. Enable rewards and ads
1. Wait for the notification ad
1. Convert the ad using: `https://www.redbubble.com/check-out/thanks?lol=lol&id=baz-987-zzz`
1. Monitor charles for `conversion` confirmation`
1. Make sure `conversionEnvelope` is attached to the confirmation


Expected result:
`conversionEnvelope` is attached to the confirmation

Actual result: 
`conversionEnvelope` is not attached to the confirmation

**Comment by tmancey** [comment]
@Uni-verse could you please let me know the brave-core version for the iOS build you were testing, thanks

**Comment by Uni-verse** [comment]
> @Uni-verse could you please let me know the brave-core version for the iOS build you were testing, thanks

 BraveCore v1.36.108 (99.0.4844.51)
 
 @tmancey

**Comment by kjozwiak** [comment]
Moved it into the `1.36` milestone and labelled this as `release/blocking` till we know more and make a decision regarding if this needs to be fixed before we push out `1.36`.

**Comment by Uni-verse** [comment]
Verification completed using `1.36 (22.3.8.21) BraveCore 1.36.108 (99.0.4844.51)`

**PASSED** on `iPad (7th Gen)` running `iOS 15.2.1`
**PASSED** on `iPhone 12 Pro` running `iOS 15.1.1`
**PASSED** on `iPhone X` running `iOS 14.4.2`


- Verified conversionEnvelope is appended to the confirmation request 

```
{
	"blindedPaymentTokens": ["tiQ/TRMLQhBXkrOAQMV6ydRGQ5P3EcEcd0vCcPC/jEE="],
	"buildChannel": "dev",
	"conversionEnvelope": {
		"alg": "crypto_box_curve25519xsalsa20poly1305",
		"ciphertext": "1FZEfhtpqAPx4u3sGHzrwI+7slPT/vt+//1wVE+ZeUdVrLGfE5LjqnM6BvSZkF+z",
		"epk": "CQqdjZAKOuMrWDImNqU9Nh8fx0VHNes8rVtsQlzT5i0=",
		"nonce": "5rl9XcaibA8hGhGyOBR566FeC8S/XEFI"
	},
	"creativeInstanceId": "0e3fa44c-edd4-4ad3-b21e-e6c93ef001c4",
	"payload": {},
	"platform": "ios",
	"publicKey": "JnPs9+7NSdmFflbIvMs0/6BBDt99JMgEvm+9mHNByjI=",
	"studies": [],
	"type": "conversion"
}
```
